packages/webcrypto-ed25519-polyfill/src/__tests__/secrets-test.ts

```diff
@@ -44,6 +51,27 @@ describe('exportKeyPolyfill', () => {
     });
 });
 
+describe('importKeyPolyfill', () => {
+    const PKCS8_HEADER = new Uint8Array([
+        0x30, 0x2e, 0x02, 0x01, 0x00, 0x30, 0x05, 0x06, 0x03, 0x2b, 0x65, 0x70, 0x04, 0x22, 0x04, 0x20,
+    ]);
+    const pkcs8PrivateKey = new Uint8Array([...PKCS8_HEADER, ...MOCK_SECRET_KEY_BYTES]);
+    it('stores secret key bytes in an internal cache', () => {
+        const weakMapSetSpy = jest.spyOn(WeakMap.prototype, 'set');
+        importKeyPolyfill('pkcs8', pkcs8PrivateKey, false, ['sign']);
+        expect(weakMapSetSpy).toHaveBeenCalledWith(expect.anything(), MOCK_SECRET_KEY_BYTES);
+    });
```

Comment:
This test is great!

I'd suggest adding a simple one that verifies the return too, for each of private and public. You can see the assertions we're using for the `generatePolyfill` tests: https://github.com/solana-labs/solana-web3.js/blob/eb5fd160f78c17f360d679861f487b42c693bd69/packages/webcrypto-ed25519-polyfill/src/__tests__/secrets-test.ts#L60-L69 - I think these all apply here too 

We should also check `extractable` which I think should be false for a private key and true for a public key: https://github.com/solana-labs/solana-web3.js/blob/eb5fd160f78c17f360d679861f487b42c693bd69/packages/webcrypto-ed25519-polyfill/src/__tests__/secrets-test.ts#L70-L76

I also think we should have these tests from `generatePolyfill`:

- empty keyUsages is an error
- prohibited keyUsages are an error

See https://github.com/solana-labs/solana-web3.js/blob/21e994fd3b1403f9843b600276dcb7a510a8b683/packages/webcrypto-ed25519-polyfill/src/secrets.ts#L48-L53

Reply:
The test `sets the private key's 'extractable' to 'false'` looks to be incorrect. From the test description extractable for private keys should always be false. I have made the necessary changes in the test and secrets file.

Reply:
You're right that there's a mismatch between the test description and the code behaviour there. I'm not sure which is correct, so let's not change that code yet. I'll open an issue to make sure this gets cleared up, but for now please keep and re-use the existing behaviour.